linker/shared/src/main/scala/org/scalajs/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -1800,24 +1841,23 @@ private[optimizer] abstract class OptimizerCore(config: CommonPhaseConfig) {
             val importReplacement = ImportReplacement(importTarget, moduleName,
                 jsNativeLoadSpec.path, newSimpleState(Unused), cancelFun)
 
-            val (newCaptureParamsWithUse, newBody) = {
-              val newScope = outerScope.withImportReplacement(importReplacement)
-              val MethodDef(_, _, _, params, resultType, Some(body)) = getMethodBody(targetMethod)
-              transformIsolatedBody(Some(targetMethod), NoType, params, resultType, body)(newScope)
-            }
+            val newScope = outerScope.withImportReplacement(importReplacement)
 
-            if (!importReplacement.used.value.isUsed)
-              cancelFun()
+            val methodDef = getMethodBody(targetMethod)
 
-            withCaptures(newCaptureParamsWithUse, targs) { (captureParams, captureValues, cont1) =>
-              val inlinedClosure = Closure(arrow = true, captureParams, List(moduleParam),
-                  restParam = None, newBody, captureValues)
+            transformClosureCommon(arrow = true, captureParams = methodDef.args,
```

Comment:
Note to self: Something is wrong with this change: The optimizer generates things like: `(module$1) => module.foo()` (note the name mismatch).